web/massastation/src/pages/Index/DashboardStation.tsx

```diff
@@ -27,20 +35,50 @@ export function DashboardStation(props: IDashboardStationProps) {
     massaPlugins,
   } = props;
 
+  const id = massaPlugins?.find(
+    (plugin: MassaPluginModel) => plugin.name === MASSA_WALLET,
+  )?.id;
+
+  const isUpdatable = massaPlugins?.find(
+    (plugin: MassaPluginModel) => plugin.name === MASSA_WALLET,
+  )?.updatable;
+
+  const [walletState, setWalletState] = useState<WalletStates>();
+
+  const { isExecuteSuccess, isExecuteLoading, updatePluginState } =
+    useUpdatePlugin(id);
+
+  useEffect(() => {
+    if (pluginWalletIsInstalled && !isUpdatable) {
+      setWalletState(WalletStates.Active);
+    } else if (pluginWalletIsInstalled && isUpdatable) {
+      setWalletState(WalletStates.Updateable);
+    } else {
+      setWalletState(WalletStates.Inactive);
+    }
+  }, [isUpdatable, pluginWalletIsInstalled]);
```

Comment:
try without a use effect and without a use state for walletState, just a const might work

Reply:
Unfortunately we need a re-render for the wallet even for this initial state... I tested logging the current states and the result was ```undefined```, ```inactive``` and finally ```active``` if you want we can sync for 5 mins to discuss this issue